packages/backend/src/helpers/http-client/index.ts

```diff
@@ -31,12 +31,14 @@ export default function createHttpClient({
   })
 
   instance.interceptors.request.use(
-    (requestConfig: AxiosRequestConfig): AxiosRequestConfig => {
+    (requestConfig: InternalAxiosRequestConfig): InternalAxiosRequestConfig => {
       const newRequestConfig = removeBaseUrlForAbsoluteUrls(requestConfig)
 
-      return beforeRequest.reduce((newConfig, beforeRequestFunc) => {
+      const result = beforeRequest.reduce((newConfig, beforeRequestFunc) => {
         return beforeRequestFunc($, newConfig)
       }, newRequestConfig)
+      // axios specifically wants InternalAxiosRequestConfig instead of AxiosRequestConfig, can utilise an assertion since both interfaces are similar
```

Comment:
```suggestion
      // axios specifically wants InternalAxiosRequestConfig instead of AxiosRequestConfig, can utilise a typecast since both interfaces are similar
```
not completely sure if this argument flies, especially as the type has been marked as internal.